src/utilities/helper.jl

```diff
@@ -73,3 +73,63 @@ function reconstruct!(r, d::MultivariateDistribution, val::AbstractVector, n::In
     r .= val
     return r
 end
+
+struct FlattenIterator{Tname, Tvalue}
+    name::Tname
+    value::Tvalue
+end
+
+Base.length(iter::FlattenIterator) where {T} = _length(iter.value)
+@inline _length(a::AbstractArray) = sum(_length, a)
```

Comment:
Are the `@inline` statements actually needed? And maybe define a special case for `AbstractArray{<:Number}` to avoid summing up 1s?

```suggestion
_length(a::AbstractArray) = sum(_length, a)
_length(a::AbstractArray{<:Number}) = length(a)
```

Reply:
Sounds good.